R/plotTemporal.R

```diff
@@ -0,0 +1,145 @@
+
+# TODO non ASCII character somewhere ...?
+
+##' Check inputs for plotTemporal function
+#'
+#' @noRd
+#' @param data list of data frames
+#' @param storm numeric or character
+#' @param var character
+#' @return NULL
+checkInputsPlotTemporal <- function(data, storm, var) {
+  
+  # Checking data input
+  stopifnot("no data found" = !missing(data))
+  stopifnot("data must be temporal series. Be sure to use data generated from temporalBehaviour() function run with product input set to 'TS'" = identical(class(data[[1]]), "list"))
+  
+  # Checking storm input
+  stopifnot("You must select one single storm" = !missing(storm))
+  if(identical(class(storm), "character")){
+    stopifnot("storm not find in data" = (storm %in% names(data)))
+  }else if(identical(class(storm), "numeric")){
+    stopifnot("storm index out of bound in data" = (storm %in% c(1, length(names(data)))))
+  }else{
+    stop("Invalid storm input")
+  }
+  
+  # Checking var input
+  stopifnot("invalid var input, must be either 'speed' or 'direction'" = 
+              var == 'speed' | var == 'direction')
+
+}
+
+
+
+
+#' Plotting wind behaviour time series and summary statistics at given point
+#' locations
+#'
+#' @param data time series generated with `temporalBehaviour` with `product=TS`
+#'   input
+#' @param storm list of characters. Storm names. The storm must be available in
+#'   `data` input. It can also be a vector of integer corresponding to the
+#'   indices of storms stored in `data`input.
+#' @param var character. Represent the type of variable to plot. Must be either
+#'   `speed` or `direction`. Default value is set to `speed`.
+#'
+#' @return null
+#' @export
+#'
+#' @examples
+#' \donttest{
+#' #Add example here
+#' }
+plotTemporal <- function(data, storm, var = 'speed'){
+  
+  checkInputsPlotTemporal(data, storm, var)
+  
+  # Filter by storm
+  sub_data <- data[storm]
+  
+  nbOfPositions <- length(sub_data[[1]])
+  
+  # Generate a sequence of colors
+  
+  # TODO pb of dependencies here...
```

Comment:
What should we do then ?

Reply:
Maybe I don't get the full problem here, but I like the way it is done with `colorRampPalette` and the number of intervals...

Reply:
Plus, I would also need it to create dynamic color palette for the default  `scalePalette`

Reply:
ok, I though `colorRampPalette` was there by default, my bad. It's fine with me, and indeed it will also work with our isse for default `scalePalette`

Reply:
Ok done

Reply:
Apparently it was grDevices::, but no need to add it in the DESCRIPTION file, I don't know why

Reply:
It is included in R-base, so no need for additional dependency.